python/kserve/kserve/kfmodel.py

```diff
@@ -110,38 +111,25 @@ def load(self) -> bool:
 
     async def preprocess(self, request: Dict) -> Union[Dict, ModelInferRequest]:
         """
-        The preprocess handler can be overridden for data or feature transformation,
-        the default implementation decodes to Dict if it is cloudevent JSON otherwise pass the data field
-        :param request: JSON Dict or CloudEvent
+        The preprocess handler can be overridden for data or feature transformation
+        the default implementation decodes to Dict if it is a binary CloudEvent
+        or gets the data field from a structured CloudEvent
+        :param request: JSON Dict
         :return: Transformed Dict|ModelInferRequest which passes to predict handler
         """
-        response = request
-
         if isinstance(request, CloudEvent):
-            response = request.data
-            if isinstance(response, bytes):
-                try:
-                    response = json.loads(response.decode('UTF-8'))
-                except (json.decoder.JSONDecodeError, UnicodeDecodeError) as e:
-                    attributes = request._attributes
-                    if "content-type" in attributes:
-                        if attributes["content-type"] == "application/cloudevents+json" or \
-                           attributes["content-type"] == "application/json":
-                            raise tornado.web.HTTPError(
-                                status_code=HTTPStatus.BAD_REQUEST,
-                                reason="Unrecognized request format: %s" % e
-                            )
+            try:
+                request = json.loads(request.data.decode('UTF-8'))
```

Comment:
I think this will fail the decoding for avro or protobuf case while the original code allows it

Reply:
Oh I see. It's a best effort try to decode it otherwise leave the request alone.

I was confused because the `DummyAvroCEModel` in `tests/test_server` overrides the `preprocess` to handle the Avro case.

I'll update